source-control/contributing/readme.md

```diff
@@ -0,0 +1,151 @@
+* [Git commit best practices](https://medium.com/@nawarpianist/git-commit-best-practices-dab8d722de99)
+
+## Contributing to an existing repository
+
+When working on an existing project, `git clone` the repository and ensure you understand the team's branch, merge and release strategy (e.g. through the projects [CONTRIBUTING.md file](https://blog.github.com/2012-09-17-contributing-guidelines/)).
+
+## Creating a new repository
+
+If you create a new repository, agree on your branch, merge and release strategies upfront and document them in your [CONTRIBUTING.md](https://blog.github.com/2012-09-17-contributing-guidelines/). Also lock the default branches and define who can approve and merge Pull Requests into your default branches.
+
+### Required files in default branch for public repositories
+
+A public repository needs to have the following files in the root directory of the default branch:
+
+* a [LICENSE](Templates/LICENSE) file
+* a [README.md](Templates/README.md) file
+* a [CONTRIBUTING.md](Templates/CONTRIBUTING.md) file
+* reference the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/)
+
+<!-- TODO: the template links to be fixed -->
+
+To start to contribute by creating your own branch, or fork the repository, and commit often - make your commit comments useful to others by including the *WHAT* and *WHY* (instead of the *HOW*).
+
+## Merge strategies
+
+Agree if you want a linear or non-linear commit history. There are pros and cons to both approaches:
+
+* Pro linear: [A tidy, linear Git history](https://www.bitsnbites.eu/a-tidy-linear-git-history/)
+* Con linear: [Why you should stop using Git rebase](https://medium.com/@fredrikmorken/why-you-should-stop-using-git-rebase-5552bee4fed1)
+
+### Approach for non-linear commit history
+
+Merging `topic` into `master`
+
+```md
+  A---B---C topic
+ /         \
+D---E---F---G---H master
+
+git fetch origin
+git checkout master
+git merge topic
+```
+
+### Two approaches to achieve a linear commit history
+
+#### Rebase topic branch before merging into master
+
+Before merging `topic` into `master`, we rebase `topic` with the   :
+
+```bash
+          A---B---C topic
+         /         \
+D---E---F-----------G---H master
+
+git fetch origin
+git rebase master topic
+git checkout master
+git merge topic
+```
+
+#### Rebase topic branch before squash merge into master
+
+[Squash merging](https://docs.microsoft.com/en-us/azure/devops/repos/git/merging-with-squash?view=azure-devops) is a merge option that allows you to condense the Git history of topic branches when you complete a pull request. Instead of adding each commit on `topic` to the history of `master`, a squash merge takes all the file changes and adds them to a single new commit on `master`.
+
+```bash
+          A---B---C topic
+         /
+D---E---F-----------G---H master
+
+Create a PR topic --> master in Azure DevOps and approve using the squash merge option
+```
+
+## Branch naming
```

Comment:
Note to other reviewers as it is hard to spot this change... this was changed from **Naming branches** to **Branch naming**
I'm good with either

Reply:
Will revert to Naming branches for consistency with Stashing changes

Reply:
Done.